src/v2/Apps/Order/Components/PriceOptions.tsx

```diff
@@ -70,20 +68,21 @@ export const PriceOptions: React.FC<PriceOptionsProps> = ({
   const asCurrency = (value: number) =>
     appendCurrencySymbol(
       value?.toLocaleString("en-US", {
-        currency: artwork?.priceCurrency!,
+        currency: order.currencyCode,
         minimumFractionDigits: 2,
         style: "currency",
       }),
-      artwork?.priceCurrency!
+      order.currencyCode!
     )
+  const offerItem = getOfferItemFromOrder(order.lineItems)
+  const isPriceRange = offerItem?.displayPriceRange
+  const listPrice = offerItem?.listPrice
 
   const getRangeOptions = () => {
-    const minPriceRange = listPrice?.minPrice?.major
-    const maxPriceRange = listPrice?.maxPrice?.major
-    const midPriceRange = Math.round(
-      (Number(minPriceRange) + Number(maxPriceRange)) / 2
-    )
-
+    const listPriceRange = listPrice as PriceRange
```

Comment:
optional: taking a look again makes sense to me to move those functions like `getRangeOptions` and `getPercentageOptions` out of the component, passing as argument Price or PriceRange variable. You wouldn't need any typecast also when calling them, for example:

```jsx
const getRangeOptions = (price: PriceRange) => {
...
}
const getPercentageOptions = (price: Price) => {
...
}

export const PriceOptions: React.FC<PriceOptionsProps> = ({
onChange,
  onFocus,
  showError,
  order,
}) => {
  ...
  const priceOptions = isPriceRange ? getRangeOptions(listPrice as PriceRange) : getPercentageOptions(listPrice as Price)
}
```
we could go further around `displayPriceRange` property, right now we check if it's true/false, we could use typeguards on `listPrice` to check if it's Price/List price, this would make us more type assertive avoiding typecast to it; not saying we should do it rn, but we should discuss more about the idea 👍